pkg/resource/endpoint/custom_update_conditions.go

```diff
@@ -51,20 +53,23 @@ func (rm *resourceManager) customUpdateConditions(
 			return false
 		}
 	}
-	if terminalCondition == nil {
-		terminalCondition = &ackv1alpha1.Condition{
-			Type:   ackv1alpha1.ConditionTypeTerminal,
-			Status: corev1.ConditionTrue,
-			Reason: aws.String("Endpoint status: Failed. Cannot be updated"),
-		}
 
-		failureReason := r.ko.Status.FailureReason
-		if failureReason != nil {
-			terminalCondition.Message = failureReason
+	if (err != nil && err == FailUpdateError) || (latestStatus != nil && *latestStatus == svcsdk.EndpointStatusFailed) {
+		// setting terminal condition since controller can no longer recover by retrying
+		if terminalCondition == nil {
+			terminalCondition = &ackv1alpha1.Condition{
+				Type: ackv1alpha1.ConditionTypeTerminal,
+			}
+			ko.Status.Conditions = append(ko.Status.Conditions, terminalCondition)
 		}
-
-		ko.Status.Conditions = append(ko.Status.Conditions, terminalCondition)
-
+		terminalCondition.Status = corev1.ConditionTrue
+		if *latestStatus == svcsdk.EndpointStatusFailed {
+			terminalCondition.Message = aws.String("Endpoint status: Failed. Cannot be updated")
```

Comment:
nit: This message feels a little clunky? Maybe `Cannot update endpoint with Failed status`?